src/Nethermind/Nethermind.Optimism/L1BlockGasInfo.cs

```diff
@@ -50,20 +67,33 @@ public L1BlockGasInfo(Block block, IOptimismSpecHelper specHelper)
 
             Memory<byte> data = depositTx.Data.Value;
 
+            _isIsthmus = _specHelper.IsIsthmus(block.Header);
             _isFjord = _specHelper.IsFjord(block.Header);
             _isPostRegolith = _specHelper.IsRegolith(block.Header);
 
-            if (_isFjord || (_isEcotone = (_specHelper.IsEcotone(block.Header) && !data[0..4].Span.SequenceEqual(BedrockL1AttributesSelector))))
+            if (_isIsthmus)
```

Comment:
Does it truly reflect https://github.com/ethereum-optimism/op-geth/blob/3d7afdc2701b74c5987e31521e2c336c4511afdf/core/types/rollup_cost.go#L381 ?

Reply:
Let me provide the reasoning. In `op-geth` equivalent there are two methods `extractL1GasParamsPostEcotone` and `extractL1GasParamsPostIsthmus`. Both of them assert the length of the data. In both versions all the fields are the same beside the two appended in the `Isthmus` format. Having that said, it can be either the selector or the length of the data to distinguish the payload. As you can see below, if the data are not matching isthmus format, we default for two new values which is equivalent to the `Econone` parsing.

Reply:
Sounds good.
Small chance still:
`if (data.Length != 164)` happens before ParsePostEcotoneBaseValues. But in _isIsthmus it parses before checking the length, which seems like if for some crazy reason length's < 164, it can fail in not expected way

Reply:
Point taken. To make it coherent with the previous, I'll check it before the call if you're ok with it. We were using default values for such case.

Reply:
Pushed updated version with a preIsthmus test data to handle this transition block.